connector-packager/connector_packager/jar_jdk_packager.py

```diff
@@ -55,7 +55,13 @@ def get_min_support_version(file_list: List[ConnectorFile], cur_min_version_tabl
                 min_version_tableau = "2020.3"
             reasons.append("Connector uses Connection Dialogs V2, which was added in the 2020.3 release")
         elif connector_file.file_type == "oauth-config":
-            if 2021.1 > float(min_version_tableau):
+            # Check to see if we're using oauthConfigId, which needs 2021.4+
+            pluginOAuthConfigRoot = ET.parse(input_dir / connector_file.file_name).getroot()
+            oauthConfigId = pluginOAuthConfigRoot.find('.//oauthConfigId')
+            if (oauthConfigId is not None):
+                min_version_tableau = "2021.4"
```

Comment:
This will work today, but to ensure we don't need to change it in the future, you'll want to check that min_version_tableau isn't already something larger than 2021.4 before making the set call.